From aba0bb881de2979a4469d66ad78f86a6e567628e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=A9my=20Bobbio?= <lunar@debian.org>
Date: Tue, 27 Aug 2013 23:53:22 +0200
Subject: [PATCH 4/4] dpkg-buildpackage: Preset build timestamp to latest
 changelog entry

dpkg-buildpackage will set the SOURCE_DATE_EPOCH environment variable to the
date of the latest entry in debian/changelog. This enables build to be easily
reproduced as the dates captured in the `.deb` archives will be deterministic.

SOURCE_DATE_EPOCH can also be externaly set before running dpkg-buildpackage.
to reproduce the build with an arbitrary date.

Closes: #759886
Closes: #759999
---
 scripts/dpkg-buildpackage.pl | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/scripts/dpkg-buildpackage.pl b/scripts/dpkg-buildpackage.pl
index 17ada97..e14cd26 100755
--- a/scripts/dpkg-buildpackage.pl
+++ b/scripts/dpkg-buildpackage.pl
@@ -439,6 +439,11 @@ if ($changedby) {
     $maintainer = mustsetvar($changelog->{maintainer}, g_('source changed by'));
 }
 
+if (!$ENV{SOURCE_DATE_EPOCH}) {
+    my $timestamp = `date -d'$changelog->{date}' +%s`;
+    chomp $timestamp;
+    $ENV{SOURCE_DATE_EPOCH} = $timestamp;
+}
 
 my @arch_opts;
 push @arch_opts, ('--host-arch', $host_arch) if $host_arch;
-- 
2.1.4

